util/Pending.h

```diff
@@ -106,15 +106,15 @@ namespace Pending {
     template <typename T>
     T& SwapPending(boost::optional<Pending<T>>& pending, T& stored) {
         if (pending) {
-            std::lock_guard<std::mutex> lock(pending->m_mutex);
+            std::scoped_lock lock(pending->m_mutex);
```

Comment:
this scoped_lock is also not strictly necessary. (see above)

Reply:
Not sure what you mean. `scoped_lock` is as far as I understand now the preferred locking mechanism over `lock_guard`. https://stackoverflow.com/questions/43019598/stdlock-guard-or-stdscoped-lock/43019620#43019620

Reply:
Sorry, i misread ("bit" tired) - thought you had switched to recursive lock with which I had a MacOS related problem. I am still unfamiliar with scoped_lock. So all my comments here are probably bullshit. Will read up.

Reply:
All this PR does should be bump version numbers in build scripts and replace the locks with the newer equivalent from C++17 std  instead of the older std or Boost.